plugins/BEdita/Core/src/Command/ObjectsDeleteCommand.php

```diff
@@ -0,0 +1,125 @@
+use Cake\Database\Expression\QueryExpression;
+use Cake\I18n\FrozenDate;
+use Cake\ORM\Locator\LocatorAwareTrait;
+
+/**
+ * ObjectsDelete command.
+ */
+class ObjectsDeleteCommand extends Command
+{
+    use LocatorAwareTrait;
+
+    /**
+     * @inheritDoc
+     */
+    public function buildOptionParser(ConsoleOptionParser $parser): ConsoleOptionParser
+    {
+        return $parser
+            ->addOption('since', [
+                'help' => 'Delete objects in trash since this date',
+                'required' => false,
+                'default' => '-1 month',
+            ])
+            ->addOption('type', [
+                'help' => 'Delete objects in trash by type',
+                'required' => false,
+                'multiple' => true,
+            ]);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(Arguments $args, ConsoleIo $io)
+    {
+        $since = $args->getOption('since');
+        $types = (array)$args->getOption('type');
+        $message = 'Deleting from trash objects, since ' . $since;
+        $message .= !empty($types) ? ', for type(s) ' . implode(',', $types) : '';
+        $io->info($message);
+        $conditions = ['deleted' => true, 'locked' => false, 'modified <' => new FrozenDate($since)];
+        $deleted = $errors = 0;
+        if (empty($types)) {
+            $types = [null];
+        }
+        foreach ($types as $objectType) {
+            foreach ($this->objectsIterator($objectType, $conditions) as $object) {
+                $this->deleteObject($io, $object, $deleted, $errors);
+            }
+        }
+        $io->success(sprintf('Deleted from trash %d objects [%d errors]', $deleted, $errors));
+        $io->info('Done');
+
+        return self::CODE_SUCCESS;
+    }
+
+    /**
+     * Delete object.
+     *
+     * @param \Cake\Console\ConsoleIo $io The console io
+     * @param \BEdita\Core\Model\Entity\ObjectEntity $object The object
+     * @param int $deleted The number of deleted objects
+     * @param int $errors The number of errors
+     * @return void
+     */
+    private function deleteObject(ConsoleIo $io, ObjectEntity $object, int &$deleted, int &$errors): void
+    {
+        try {
+            $io->verbose(sprintf('Deleting object %s', $object->id));
+            $object->getTable()->deleteOrFail($object);
+            $deleted++;
+        } catch (\Throwable $e) {
+            $io->error(sprintf('Error deleting object %s: %s', $object->id, $e->getMessage()));
+            $errors++;
+        }
+    }
+
+    /**
+     * Get objects as iterable.
+     *
+     * @param string|null $type The object type
+     * @param array $conditions The conditions
+     * @return iterable
+     */
+    private function objectsIterator(?string $type, array $conditions): iterable
+    {
+        $table = $this->fetchTable('Objects');
+        $query = empty($type) ? $table->find() : $table->find('type', [$type]);
+        $query = $query->where($conditions)->limit(200);
```

Comment:
Maybe it's better to sort by id asc since it is used to fetch results where  `'id' > $lastId` in the while cycle below.